src/games/strategy/engine/EngineVersion.java

```diff
@@ -1,7 +1,33 @@
 package games.strategy.engine;
 
+import games.strategy.engine.config.GameEngineProperty;
+import games.strategy.engine.config.PropertyReader;
 import games.strategy.util.Version;
 
 public class EngineVersion {
-  public static final Version VERSION = new Version(1, 8, 0, 9);
+  private final Version version;
+  private final String exactVersion;
+
+  public EngineVersion(PropertyReader propertyReader) {
```

Comment:
PropertyReader here is dependency injection. The test case for this is very simple because we don't have to write to a file to test this. A first direction was to pass in a File to the constructor, parse it to get the EngineVersion, but that made a difficult. Either we need static initialization code somewhere (which is basically what we have in ClientContext), or each client needs to have a `PropertyReader` object it can pass in when it wants ot use an EngineVersion.